craft_parts/plugins/maven_plugin.py

```diff
@@ -206,5 +207,15 @@ def _create_settings(settings_path: Path) -> None:
 
 
 def _get_no_proxy_string() -> str:
-    no_proxy = [k.strip() for k in os.environ.get("NO_PROXY", "localhost").split(",")]
+    no_proxy = []
+
+    for address in os.environ.get("NO_PROXY", "localhost").split(","):
+        if address:
+            no_proxy.append(address.strip())
+
+    # backwards compatibility
```

Comment:
I don't understand why we're mixing functional changes with the linting work. Can't we just keep the original code?

Reply:
Sure, but I changed it because the linting error:
```
craft_parts/plugins/maven_plugin.py:210:51: SIM112 Use capitalized environment variable `NO_PROXY` instead of `no_proxy`
```